bot/seasons/evergreen/minesweeper.py

```diff
@@ -0,0 +1,266 @@
+
+    def __init__(self, bot: commands.Bot) -> None:
+        self.games: GamesDict = {}  # Store the currently running games
+
+    @commands.group(name='minesweeper', aliases=('ms',), invoke_without_command=True)
+    async def minesweeper_group(self, ctx: commands.Context):
+        """Commands for Playing Minesweeper"""
+        await ctx.send_help(ctx.command)
+
+    @staticmethod
+    def get_neighbours(x: int, y: int) -> typing.Generator[typing.Tuple[int, int], None, None]:
+        """Get all the neighbouring x and y including it self."""
+        for x_ in [x - 1, x, x + 1]:
+            for y_ in [y - 1, y, y + 1]:
+                if x_ != -1 and x_ != 10 and y_ != -1 and y_ != 10:
+                    yield x_, y_
+
+    def generate_board(self, bomb_chance: float) -> GameBoard:
+        """Generate a 2d array for the board."""
+        board: GameBoard = [
+            [
+                "bomb" if random() <= bomb_chance else "number"
+                for _ in range(10)
+            ] for _ in range(10)
+        ]
+        for y, row in enumerate(board):
+            for x, cell in enumerate(row):
+                if cell == "number":
+                    # calculate bombs near it
+                    bombs = 0
+                    for x_, y_ in self.get_neighbours(x, y):
+                        if board[y_][x_] == "bomb":
+                            bombs += 1
+                    board[y][x] = bombs
+        return board
+
+    @staticmethod
+    def format_for_discord(board: GameBoard) -> str:
+        """Format the board as a string for Discord."""
+        discord_msg = (
+            ":stop_button:    :regional_indicator_a::regional_indicator_b::regional_indicator_c:"
+            ":regional_indicator_d::regional_indicator_e::regional_indicator_f::regional_indicator_g:"
+            ":regional_indicator_h::regional_indicator_i::regional_indicator_j:\n\n"
+        )
+        rows = []
+        for row_number, row in enumerate(board):
+            new_row = f"{MESSAGE_MAPPING[row_number + 1]}    "
+            new_row += "".join(MESSAGE_MAPPING[cell] for cell in row)
+            rows.append(new_row)
+
+        discord_msg += "\n".join(rows)
+        return discord_msg
+
+    @minesweeper_group.command(name="start")
+    async def start_command(self, ctx: commands.Context, bomb_chance: float = .2) -> None:
+        """Start a game of Minesweeper."""
+        if ctx.author.id in self.games:  # Player is already playing
+            await ctx.send(f"{ctx.author.mention} you already have a game running!", delete_after=2)
+            await ctx.message.delete(delay=2)
+            return
+
+        # Add game to list
+        board: GameBoard = self.generate_board(bomb_chance)
+        revealed_board: GameBoard = [["hidden"] * 10 for _ in range(10)]
+
+        if ctx.guild:
+            await ctx.send(f"{ctx.author.mention} is playing Minesweeper")
+            chat_msg = await ctx.send(self.format_for_discord(revealed_board))
```

Comment:
We'll need some kind of header line for this so the board gets rendered properly:

![image](https://user-images.githubusercontent.com/5323929/62986989-e193fa80-be0b-11e9-9bfc-7bdb66c2f89e.png)


Something like `"Here's your board!\n"` should be fine (leading whitespace is stripped by Discord)

Reply:
Never noticed that, as its only like that on compact mode. I can do it tomorrow as its getting late.